File: Python/3042CountPrefixAndSuffixPairs1.py
class Solution(object):
    def countPrefixSuffixPairs(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        count = 0
        for i in range(len(words)):
            for j in range(i + 1, len(words)):
                if self.isPrefixAndSuffix(words[i], words[j]):
                    count += 1
        return count

    def isPrefixAndSuffix(self, str1, str2):
        if (len(str1) > len(str2)):
            return False
        for i in range(len(str1)):
            if str1[i] != str2[i]:
                return False
        for i in range(len(str1)):
            if str1[i] != str2[len(str2) - len(str1) + i]:
                return False
        return True

File: C++/3042CountPrefixAndSuffixPairs1.cpp
//
//  3042CountPrefixAndSuffixPairs1.cpp
//  Testing C++
//
//  Created by Cem Beyenal on 1/8/25.
//
#include <iostream>  // For input/output
#include <vector>    // For std::vector
#include <string>    // For std::string
using namespace std;

class Solution {
public:
    int countPrefixSuffixPairs(vector<string>& words) {
        int count = 0;
        for (int i = 0; i < words.size(); i++) {
            for (int j = i + 1; j < words.size(); j++) {
                if (isPrefixAndSuffix(words[i], words[j])) {
                    count++;
                }
            }
        }
        return count;
    }

    bool isPrefixAndSuffix(string str1, string str2) {
        bool isPrefix = true, isSuffix = true;
        int str2Len = str2.length();
        if (str2.size() < str1.size()) {
            return false;
        }
        for (int i = 0; i < str1.length(); i++) {
            if (str1[i] != str2[i]) {
                isPrefix = false;
                break;
            }
        }
        for (int i = 0; i < str1.length(); i++) {
            if (str1[i] != str2[str2Len - str1.length() + i]) {
                isSuffix = false;
                break;
            }
        }
        return isSuffix && isPrefix;
    }
};

